### PR TITLE
Feature/sg 1155 support pil

### PR DESCRIPTION
**Pull request by Louis-Dupont**
## What ?
Make the output_mapper support loading images from List of PIL image, or List of string.

## Why ?
When you try to use torchvision datasets, the output of the dataset is PIL images. If the user passes a Dataloader, the output will be a list of PIL images, which is currently not supported. The same issue could happen if the dataset outputs a string, then we would not support the Dataloader.

## Extra
I also added some extra fixes to make the unit tests properly work with the CI.

### CODE REVIEW

File: src/data_gradients/dataset_adapters/output_mapper/dataset_output_mapper.py [hunk 1]
```diff
@@ -82,13 +93,41 @@ def _get_labels_extractor(self, data: SupportedData) -> Callable[[SupportedData]
...
+        elif isinstance(data, np.ndarray):
+            return torch.from_numpy(data)
+        elif isinstance(data, PIL.Image.Image):
+            return torch.from_numpy(np.array(data))
+        elif isinstance(data, str):
+            with PIL.Image.open(data) as img:
+                return torch.from_numpy(np.array(img))
+        elif np.isscalar(data):
+            return torch.tensor(data)
+        elif isinstance(data, list):
```
Comment (BloodAxe) [inline src/data_gradients/dataset_adapters/output_mapper/dataset_output_mapper.py:120]: + tuple ?
Also, I'm not sure whether this may happen or not, what if data will be ListConfig or DictConfig (OmegaConf containers)?

**Review by BloodAxe — DISMISSED** [review]
LGTM

File: src/data_gradients/dataset_adapters/output_mapper/dataset_output_mapper.py (hunk 1 above)
Comment (Louis-Dupont) [reply src/data_gradients/dataset_adapters/output_mapper/dataset_output_mapper.py]: Good point, I added support for `Sequence` https://github.com/Deci-AI/data-gradients/pull/199/commits/0c3dd404317be795611e8591a0fc74169e5c238f

**Review by BloodAxe — APPROVED** [review]
LGTM
